Library/Homebrew/keg.rb

```diff
@@ -157,6 +158,25 @@ def self.find_some_installed_dependents(kegs)
     nil
   end
 
+  # Finds kegs that were installed as dependencies of other kegs,
+  # but that are no longer required by any explicitly-installed kegs.
+  def self.orphaned
+    all.select(&:orphaned?)
+  end
+
+  def orphaned?
+    return false if might_have_been? :installed_on_request
+    installed_dependents.none? { |d| d.might_have_been? :installed_on_request }
+  end
+
+  def might_have_been?(key)
+    Tab.for_keg(self).absent_or_truthy?(key)
+  end
+
+  def definitely_was_not?(key)
+    Tab.for_keg(self).present_and_falsey?(key)
```

Comment:
It's not — I put it there for consistency with `#might_have_been?`, its inverse.